### ISSUE TITLE
Error in calculation of marginal work prices for 2015

### ISSUE THREAD

**Issue by wagnertimo** [wagnertimo/rmarketcrawlR#8]
An error occurred while calculating the marginal work price out of the approximated 1min calls of 2015 from the function `approximateCallsInRecursion()`. 

The parallel computational function `calcMarginalWorkPrices()` triggers an error indicating missing rows. The problem is that in parallel mode, debugging is hard since tracing back the error is impossible.

**LOG:**
```
Fehler in { : 
  task 1 failed - "arguments imply differing number of rows: 0, 1"
Called from: e$fun(obj, substitute(ex), parent.frame(), e$data)
```

**Comment by wagnertimo** [comment]
Investigations:

1. **Looking for missing values in the approximated 2015 calls** --> no missing values
  a. It was noticed that on `2015-03-20` a daily auction happend such that the data is no longer weeklys. But this should not cause errors since the function `matchAuctionsWithCalls()` filters the dates, times and direction in the auctions dataset.

2. **Splitting the approximated calls on a monthly base to narrow the error** --> every month causes the same error. Monthly data is correct.

3. **Downloading monthly data and calculating everything monthly** 
--> since now: January, February caused no problem.
--> This worked fine. Everything is ok.

--> Since *manual* approach worked. I compared the monthly approximated calls from the newly *manually* downloaded data. Except January there is a difference in `TZ` variable when calling `all.equal()`. It seems that the *old* call and need data on server is maybe wrong

New test approach: *Everything new for 2015 on server*

**Comment by wagnertimo** [comment]
**New bug noticed!:**

> Side effect --> the function `getAuctionIds()` calculates the difference in weeks between the `startDate` and `endDate` to set the number of look ups for the auctionIDs (this is equal to the `auctionIds` array length). The problem is that e.g. in `2015-03-20` a daily auction occurs in a seperate row which leads to one week less for the year 2015.

--> New issue opened

**Comment by wagnertimo** [comment]
As suggested I re-crawled the auctions (with the bug fixes) and calls and needs again and then calculated the approximated calls for 2015. The `all.equals()` function indicated no difference in the approximated calls data set and the old one (where the bug occured).

But in the end I was able to realize the marginal work price calculation.

Would have been strange since it manually worked. --> Issue hereby closed
